src/components/menu/menu.tsx

```diff
@@ -0,0 +1,229 @@
+import { Component, Element, h, Host, Listen, Method, Prop, State, Watch } from "@stencil/core";
+import { focusElement, focusElementInGroup, slotChangeGetAssignedElements } from "../../utils/dom";
+import {
+  componentLoaded,
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent
+} from "../../utils/loadable";
+import { connectLocalized, disconnectLocalized, LocalizedComponent } from "../../utils/locale";
+import {
+  connectMessages,
+  disconnectMessages,
+  setUpMessages,
+  T9nComponent,
+  updateMessages
+} from "../../utils/t9n";
+import { MenuMessages } from "./assets/menu/t9n";
+
+type Layout = "horizontal" | "vertical";
+
+@Component({
+  tag: "calcite-menu",
+  styleUrl: "menu.scss",
+  shadow: true,
+  assetsDirs: ["assets"]
+})
+export class CalciteMenu implements LoadableComponent, LocalizedComponent, T9nComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteMenuElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Specifies the layout of the component.
+   */
+  @Prop({ reflect: true }) layout: Layout = "horizontal";
+
+  /**
+   * Specifies accessible label for the component.
+   */
+  @Prop() label!: string;
+
+  /**
+   * @internal
+   */
+  @Prop() role = "menubar";
```

Comment:
Should this use the global attributes helper?

Reply:
This is an internal prop and we are not mutating this after initial render. I dont think globalAttributes helper is required here.